pipeline_steps/common.groovy

```diff
@@ -1765,6 +1766,85 @@ void createComponentGateTrigger(String name, String repoUrl, String jiraProjectK
   }
 }
 
+void createComponentPreReleaseJobs(String name, String repoUrl, List releases, String jiraProjectKey){
+  String repo_dir = "${WORKSPACE}/rpc-gating-master"
+  String filename = "${name}.yml".replace("-", "_")
+  String projectsFile = "${repo_dir}/rpc_jobs/${filename}"
+  String jjb
+
+  dir(repo_dir) {
+    // NOTE(mattt): We re-clone rpc-gating here because the existing clone
+    // has cruft in it which we don't want to have committed in our
+    // subsequent PR
+    git branch: 'master', url: 'https://github.com/rcbops/rpc-gating'
+
+    Integer projectNotExists = sh(
+      returnStatus: true,
+      script: """#!/bin/bash -xe
+        grep -s 'RE-Release-PR_{repo}-{BRANCH}' ${projectsFile}
+      """
+    )
+
+    // NOTE(mattt): If projectsFile does not exist, projectNotExists will get
```

Comment:
Why does this skip if the file doesn't exist? That seems like it would prevent this from being applied to new projects. How will this play with `createComponentGateTrigger`? Will the pull requests conflict?

Reply:
I _think_ my reasoning was that if the project has no jobs defined in rpc-gating yet, then it'd be unlikely that they're ready to have this job configured.

Good question about `createComponentGateTrigger`. I tested the change in isolation to avoid creating a bunch of PRs across repos, so didn't witness if this would conflict. I imagine it will, resulting in the last to be merged PR needing a rebase.

Reply:
@git-harry I guess I can merge `createComponentGateTrigger` and `createComponentPreReleaseJobs` together so we create a single PR ?